include/Jit/LLILCJit.h

```diff
@@ -174,6 +180,32 @@ class NullResolver : public llvm::RuntimeDyld::SymbolResolver {
   }
 };
 
+class ObjectLoadListener {
+public:
+  ObjectLoadListener(LLILCJitContext *Context) { this->Context = Context; }
+
+  template <typename ObjSetT, typename LoadResult>
+  void operator()(llvm::orc::ObjectLinkingLayerBase::ObjSetHandleT ObjHandles,
+                  const ObjSetT &Objs, const LoadResult &LoadedObjInfos) {
+    for (const std::unique_ptr<object::ObjectFile> &PObj : Objs) {
```

Comment:
Nit: I'd still prefer `const auto &PObj` here, since we're going to the trouble of abstracting the set type and pointer kind to match the abstraction in the calling code.